### PR TITLE
Fix formatting and parsing changes

### PR DESCRIPTION
**Pull request by osowskit**
* Switch indentation back to two spaces
* Switch `verify_record` to use the function parameter `challenge`
* Switch `PublicSuffix` to use `parse` instead of `domain`

### CODE REVIEW

File: dnsimple_hook.rb [hunk 1]
```diff
@@ -13,30 +14,34 @@
...
+def verify_domain_control(account_id, domain_name)
   domains = @client.domains.list_domains(account_id).data
-
   domains.each do |domain|
-    domain_name = domain.name
-    if full_domain_name.include? domain_name
-      return_val["domain_name"] = domain_name
-      return_val["subdomain_name"] = full_domain_name.chomp(".#{domain_name}")
+    dns_simple_domain_name = domain.name
+    if dns_simple_domain_name.eql? domain_name
```
Comment (jbjonesjr) [inline dnsimple_hook.rb:23]: why assign just to use in only once in the next line. While maybe a bit more readable, i don't think it's worth it.

File: dnsimple_hook.rb [hunk 2]
```diff
@@ -13,30 +14,34 @@
...
-     return resp.strings[0] == txt_challenge      
-    }
+# This function returns the result of a specific text string(the challenge) 
+# being stored in a DNS TXT record for a domain(challenge_fqdn)
+def verify_record(challenge_fqdn, challenge)
+  found_record = false
+
+  @dns.each_resource(challenge_fqdn, Resolv::DNS::Resource::IN::TXT) { |resp|
+    if resp.strings[0] == challenge
+      found_record = true
```
Comment (jbjonesjr) [inline dnsimple_hook.rb]: maybe change this to `return true` so we can short-circuit future iterations of the loop?

File: dnsimple_hook.rb [hunk 3]
```diff
@@ -13,30 +14,34 @@
...
+    end
+  }
+  return found_record
```
Comment (jbjonesjr) [inline dnsimple_hook.rb]: this can be `return false` since we short circuited above

Comment (jbjonesjr) [reply dnsimple_hook.rb]: **NOTE:** With these changes, we can :fire: the `found_record` attribute

**Review by jbjonesjr — CHANGES_REQUESTED** [review]
just minor optimizations suggested, rest of the code looks SOLID. I found the parse and challenge  bugs myself (and pushed them to the main branch before i saw this :( )

**Comment by jbjonesjr** [comment]
Feel free to ignore my changes though, they are more preference pieces, not problems

File: dnsimple_hook.rb (hunk 2 above)
Comment (osowskit) [reply dnsimple_hook.rb]: Agreed ❤️. Updated.

File: dnsimple_hook.rb (hunk 3 above)
Comment (osowskit) [reply dnsimple_hook.rb]: Done.

**Review by jbjonesjr — APPROVED** [review]

**Comment by jbjonesjr** [comment]
Approving, not sure if you want to do anything with the domain_name variables and such (they seem a bit overloaded).

Instead of my suggestion from last night, we may be better off giving `domain_name` a more descriptive name to start with

**Comment by jbjonesjr** [comment]
Think we should merge and consider this an MVP.

**Comment by osowskit** [comment]
Sure thing. Thanks for the reviews

**Pull request merged**